PONDER_README.md

```diff
@@ -0,0 +1,40 @@
+The ponder product uses this branch.
+
+# Adding new commits
+Open a pull request within this repository. Some members of Ponder can approve those requests and
+merge them. **MAKE SURE THAT YOUR PULL REQUEST IS SAFE FOR THE PONDER PRODUCT BEFORE YOU MERGE IT.**
+
+In the PR description, include the tag `[upstream]` or `[no upstream]` to indicate whether we should
+incorporate some version of this commit into upstream modin at modin-project/modin.
+
+TODO(mvashishtha): Add a pre-commit check for the tag.
+
+# History
+The commits through acc95c14a911e4a6195a94039d5e930155269d97 were originally in @devin-petersohn's
+[service/init-prod](https://github.com/devin-petersohn/modin/tree/service/init-prod/modin) branch.
+@vnlitvinov rebased those commits on modin release 0.18.1 and reordered and split them so that the
+commits starting from 0.18.1 would go in the order 1) commits that we want to eventually upstream
+into modin-project/modin and 2) commits that we don't want to upstream, e.g. fixes that are
+relevant to the ponder product but not to the rest of the Modin world. Looking at the
+[history](https://github.com/ponder-org/modin-public/commits/ponder-on-modin-0-18-1), you can see
+that:
+
+- 753fd91b0e5a2f79b6bad0c9f36e6e303fa5602b was the last commit in modin 0.18.1
+- aff32bef3821da310a723e4e89e1d7be297907cd was the first upstreamable commit from service/init-prod
+- e2f4235c7ccdc139766541a4b3ef8fea608108bb was the first non-upstreamable commit
+
+The original service/init-prod has been archived in the branch
+[ponder-on-modin-0a2c0de4451f7e2e8f337a9478d7595473aa348e](https://github.com/ponder-org/modin-public/tree/ponder-on-modin-0a2c0de4451f7e2e8f337a9478d7595473aa348e)
+
+# Doing the next modin rebase
+For the next rebase, we need to once again
+
+1) create a copy of this branch
+1) Rebase onto a newer release of modin
+1) drop the commits that have already been upstreamed or that are no longer useful (e.g. because
+   now the client query compiler is no longer in modin)
```

Comment:
```suggestion
1) drop the commits that have already been upstreamed or that are no longer useful (e.g. because
   now the client query compiler is no longer in modin, we can drop all commits touching the client
   query compiler)
```